enumerables.rb

```diff
@@ -0,0 +1,97 @@
+module Enumerable
+  def my_each()
+    i = -1
+    length.times do
+      i += 1
+      yield self[i]
+    end
+  end
+
+  def my_each_with_index()
+    i = -1
+    length.times do
+      i += 1
+      yield self[i], i
+    end
+  end
+
+  def my_select(&block)
+    new_array = []
+    my_each do |item|
+      new_array << item if block.call(item) == true
+    end
+    new_array
+  end
+
+  def my_all?(&block)
+    new_array = []
```

Comment:
* [ ] `my_all?`  identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

```p %w[ant bear cat].my_all?(/t/)```  should return  #=> false
``` p [1, 2i, 3.14].my_all?(Numeric)``` should return   #=> true
``` p [nil, true, 99].my_all?  should ``` should return  #=> false